drivers/i2c/Kconfig.dw

```diff
@@ -4,11 +4,17 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+# This option signifies whether DesignWare I2C compatible HW is
+# available
+config HAS_I2C_DW
+	bool
+
 menuconfig I2C_DW
 	bool "Design Ware I2C support"
 	select HAS_DTS_I2C
+	depends on HAS_I2C_DW
```

Comment:
Wouldn't it be better to depend on SoC families instead? To stay consistent with other drivers' Kconfigs (see for instance esp32, nrfx, stm32, also SPI driver).

Reply:
Discussed offline.

Doing it like this allows Kconfig.dw to be used with out-of-tree HW support.